bc/channel/management/commands/post.py

```diff
@@ -1,49 +1,95 @@
+from django.conf import settings
 from django.core.management.base import BaseCommand
-from prettytable import PrettyTable
+from django_rq.queues import get_queue
+from rq import Retry
 
-from bc.channel.models import Channel
-from bc.channel.utils.connectors.masto import MastodonConnector
-from bc.channel.utils.connectors.twitter import TwitterConnector
+from bc.channel.models import Channel, Group
+from bc.core.utils.commands import (
+    show_all_channels_table,
+    show_channel_groups_table,
+)
+
+queue = get_queue("default")
+
+
+def handle_post_command(
+    ids: list[int], mapping: dict[int, Channel | Group], text: str
+) -> None:
+    """
+    Enqueues a job to create a post for each channel in the list of ids provided
+    by the user.
+
+    Args:
+        ids (list[int]): list of the id from the user's input.
+        mapping (dict[int, Channel  |  Group]): Mapping of objects in the table.
+        text (str): Text to include in the posts.
+
+    Raises:
+        ValueError: if the provided input is not in the mapping variable.
+    """
+    for record_id in ids:
+        record = mapping.get(record_id)
+
+        if record is None:
+            raise ValueError(f"No channel {record_id}")
+
+        if isinstance(record, Group):
+            for channel in record.channels.all():
+                api = channel.get_api_wrapper()
+                queue.enqueue(
+                    api.add_status,
+                    text,
+                    None,
+                    retry=Retry(
+                        max=settings.RQ_MAX_NUMBER_OF_RETRIES,
+                        interval=settings.RQ_RETRY_INTERVAL,
+                    ),
+                )
```

Comment:
Didn't we create a helper for this loop?

Reply:
We included a helper to refactor a loop similar to this one but it's not the same (even though it iterates over the channel list). 

In the `enqueue_posts_for_new_case` helper, We use one of the templates to compute the content of the post(We don't need to provide a message), but the loop in the `post` command is different because We use the plain text provided by the user through the command and create a new status in each channel (We don't need to use templates or create thumbnails)

I'm going to add a helper to refactor this loop.